### PR TITLE
Ansible / yamllint CI processing

### PR DESCRIPTION
**Pull request by stefancrain**
- Added a Makefile to pretty and lint locally
- Corrected yaml formatting to pass yamllint
- Updated a couple of Ansible files to correctly follow links

### CODE REVIEW

File: ansible/.yamllint [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  commas: {max-spaces-after: -1, level: error}
+  comments: disable
+  comments-indentation: disable
+  document-start: enable
+  empty-lines: {max: 3,  max-start: 2, max-end: 2,level: error}
+  hyphens: {level: error}
+  indentation: disable
+  key-duplicates: enable
+  line-length: disable
+  new-line-at-end-of-file: disable
```
Comment (xunholy) [inline ansible/.yamllint:23]: I actually live by the `new-line-at-end-of-file` principle, if you can enable it that would be awesome.

File: ansible/Makefile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+			help_split=($$help_line) ; \
+			help_command=`echo $${help_split[0]} | sed -e 's/^ *//' -e 's/ *$$//'` ; \
+			help_info=`echo $${help_split[2]} | sed -e 's/^ *//' -e 's/ *$$//'` ; \
+			printf '\033[36m'; \
+			printf "%-30s %s" $$help_command ; \
+			printf '\033[0m'; \
+			printf "%s\n" $$help_info; \
+	done
+
+install: ##install requirements
```
Comment (xunholy) [inline ansible/Makefile:19]: I don't see a corresponding file for `requirements.yml` or is this just for in the future use case where that might be present?

File: ansible/playbooks/master.yml [hunk 3]
```diff
@@ -6,9 +6,15 @@
...
+    - ../roles/masters
+    - {
+        role: ../roles/initial-master,
+        when: "groups['masters'][0] == inventory_hostname",
+      }
+    - { role: ../roles/cni, when: "groups['masters'][0] == inventory_hostname" }
+    - {
+        role: ../roles/other-masters,
+        when: "groups['masters'][0] != inventory_hostname",
+      }
```
Comment (xunholy) [inline ansible/playbooks/master.yml:20]: Strange it puts multiline for the 1st and 3rd but not the second, anything we can tweak to make it consistent?

**Review by xunholy — COMMENTED** [review]
Looking really good!

File: ansible/playbooks/master.yml (hunk 3 above)
Comment (stefancrain) [reply ansible/playbooks/master.yml:20]: What a weird convention, looks to be line length dependent. I'll see what I can do

**Comment by stefancrain** [comment]
So much has changed, will attempt this in new PR if needed.

**Comment by xunholy** [comment]
@stefancrain a lot has been done and improved! Will sync up on discord when you have time.

**Pull request closed without merging**